### PR TITLE
Added the disable button option to connect to database

### PR DESCRIPTION
**Pull request by VivekY1098**
Fixes #<issue_number_goes_here>
Screenshot 2024-12-10 at 9 48 26 AM

> It's a good idea to open an issue first for discussion.

- [ ] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: ui/src/app/components/home/home.component.ts [hunk 1]
```diff
@@ -25,4 +38,16 @@ export class HomeComponent implements OnInit {
       this.router.navigate(['/prepare-migration'])
     }
   }
+
+  connectToDatabase(){
+    if(this.isOfflineStatus){
+      this.dialog.open(InfodialogComponent, {
+        data: { message: "Please configure spanner project id and instance id to proceed", type: 'error', title: 'Configure Spanner' },
```
Comment (taherkl) [inline ui/src/app/components/home/home.component.ts:44]: Can you add a screenshot of the pop-up or short video of how will it look like?

File: ui/src/app/components/home/home.component.ts [hunk 2]
```diff
@@ -4,18 +4,31 @@ import { Router } from '@angular/router';
...
   constructor(private dialog: MatDialog,
     private router: Router,
-    private healthCheckService: BackendHealthService) { }
+    private data: DataService,
+    private healthCheckService: BackendHealthService) {
+
+     }
 
   ngOnInit(): void {
+
```
Comment (taherkl) [inline ui/src/app/components/home/home.component.ts]: nit: Remove extra line

**Comment by taherkl** [comment]
@VivekY1098 What happens if u edit an invalid configuration on the next page? i.e. Page where u input the Source DB details? please check this scenario

**Comment by taherkl** [comment]
@VivekY1098 Please add UTs for the new functions added

File: ui/src/app/components/home/home.component.ts (hunk 2 above)
Comment (VivekY1098) [reply ui/src/app/components/home/home.component.ts]: resolved the comment

**Review by taherkl — APPROVED** [review]

**Pull request merged**